sys/include/net/sock.h

```diff
@@ -99,13 +98,80 @@ extern "C" {
 #endif
 
 /**
- * @brief   Common flags for @ref net_conn
+ * @brief       Common flags for @ref net_conn
+ * @name        net_sock_flags
  * @{
  */
-#define SOCK_FLAGS_REUSE_EP     (0x00000001)    /**< allow to reuse end point on bind */
+#define SOCK_FLAGS_REUSE_EP     (0x0001)    /**< allow to reuse end point on bind */
 /** @} */
 
 
+/**
+ * @brief   Special netif ID for "any interface"
+ * @todo    Use an equivalent defintion from PR #5511
+ */
+#define SOCK_ADDR_ANY_NETIF (0)
+
+/**
+ * @brief   Address to bind to any IPv4 address
+ */
+#define SOCK_IPV4_EP_ANY(netif) { .family = AF_INET, \
```

Comment:
static initializers do implicit zeroing of all other members., no need for explicitly zeroing .addr.
There's no "binding to interfaces" in IPv4, only "binding to IP address (at least on all the other systems we know), so you can drop the netif parameter.